src/extractor/edge_based_graph_factory.cpp

```diff
@@ -417,56 +426,72 @@ void EdgeBasedGraphFactory::GenerateEdgeExpandedEdges(
                 BOOST_ASSERT(!edge_data2.reversed);
 
                 // the following is the core of the loop.
-                unsigned distance = edge_data1.distance;
-                if (m_traffic_lights.find(node_v) != m_traffic_lights.end())
-                {
-                    distance += profile_properties.traffic_signal_penalty;
-                }
-
-                const int32_t turn_penalty =
-                    scripting_environment.GetTurnPenalty(180. - turn.angle);
-                const auto turn_instruction = turn.instruction;
-
-                if (turn_instruction.direction_modifier == guidance::DirectionModifier::UTurn)
-                {
-                    distance += profile_properties.u_turn_penalty;
-                }
-
-                distance += turn_penalty;
 
+                // First write out the meta information of a turn edge
                 BOOST_ASSERT(m_compressed_edge_container.HasEntryForID(edge_from_u));
                 original_edge_data_vector.emplace_back(
                     m_compressed_edge_container.GetPositionForID(edge_from_u),
                     edge_data1.name_id,
                     turn.lane_data_id,
-                    turn_instruction,
+                    turn.instruction,
                     entry_class_id,
                     edge_data1.travel_mode);
-
                 ++original_edges_counter;
-
                 if (original_edge_data_vector.size() > 1024 * 1024 * 10)
                 {
                     FlushVectorToStream(edge_data_file, original_edge_data_vector);
                 }
 
+                // compute weight and duration penalties
+                TurnPenalty weight_penalty = 0;
+                TurnPenalty duration_penalty = 0;
+                if (m_traffic_lights.find(node_v) != m_traffic_lights.end())
+                {
+                    // FIXME we can't assume this here!
+                    weight_penalty += profile_properties.traffic_signal_penalty;
+                    duration_penalty += profile_properties.traffic_signal_penalty;
+                }
+
+                // MKR if (use_turn_function)
```

Comment:
Probably a left-over. ProcessTurn checks internally if this function exists. I think the if-encapsulation can be removed.